### PR TITLE
feat: #1042 Allow configure db ConnMaxLifetime & ConnMaxIdleTime

### LINKED ISSUE #1042: allow configure db `ConnMaxLifetime` & `ConnMaxIdleTime`
improve https://github.com/scroll-tech/scroll/pull/1032

### PR DESCRIPTION
**Pull request by jeswinsimon**
### Purpose or design rationale of this PR

Allow `connMaxLifetime` and `connMaxIdleTime` to be configured for database connections.

Fixed #1042.

### feat: https://github.com/scroll-tech/scroll/issues/1042 Allow configure db ConnMaxLifetime & ConnMaxIdleTime

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [ ] build: Changes that affect the build system or external dependencies (example scopes: yarn, eslint, typescript)
- [ ] ci: Changes to our CI configuration files and scripts (example scopes: vercel, github, cypress)
- [ ] docs: Documentation-only changes
- [x] feat: A new feature
- [ ] fix: A bug fix
- [ ] perf: A code change that improves performance
- [ ] refactor: A code change that doesn't fix a bug, or add a feature, or improves performance
- [ ] style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
- [ ] test: Adding missing tests or correcting existing tests

### Deployment tag versioning

Has `tag` in `common/version.go` been updated or have you added `bump-version` label to this PR?

- [x] No, this PR doesn't involve a new deployment, git tag, docker image tag
- [ ] Yes

### Breaking change label

Does this PR have the `breaking-change` label?

- [x] No, this PR is not a breaking change
- [ ] Yes

### CODE REVIEW

**Comment by jeswinsimon** [comment]
@HAOYUatHZ Please review.

File: bridge-history-api/conf/config.json [hunk 1]
```diff
@@ -40,7 +40,9 @@
 		"dsn": "postgres://postgres:123456@localhost:5444/test?sslmode=disable",
 		"driverName": "postgres",
 		"maxOpenNum": 200,
-		"maxIdleNum": 20
+		"maxIdleNum": 20,
+		"maxLifetime": 600,
+		"maxIdleTime": 300
```
Comment (Flukena) [inline bridge-history-api/conf/config.json:48]: you can create static value ?

File: common/database/db.go [hunk 2]
```diff
@@ -70,8 +70,17 @@ func InitDB(config *Config) (*gorm.DB, error) {
 		return nil, pingErr
 	}
 
-	sqlDB.SetConnMaxLifetime(time.Minute * 10)
-	sqlDB.SetConnMaxIdleTime(time.Minute * 5)
+	if (config.MaxLifetime != 0) {
+		sqlDB.SetConnMaxLifetime(time.Second * time.Duration(config.MaxLifetime))
```
Comment (HAOYUatHZ) [inline common/database/db.go:74]:
```suggestion
		sqlDB.SetConnMaxLifetime(defaultMaxLifetime)
```

File: common/database/db.go [hunk 3]
```diff
@@ -70,8 +70,17 @@ func InitDB(config *Config) (*gorm.DB, error) {
...
+	} else {
+		sqlDB.SetConnMaxLifetime(time.Minute * 10)
```
Comment (0xmountaintop) [inline common/database/db.go]:
```suggestion
		sqlDB.SetConnMaxLifetime(defaultConnMaxLifetime)
```

File: common/database/db.go [hunk 4]
```diff
@@ -70,8 +70,17 @@ func InitDB(config *Config) (*gorm.DB, error) {
...
+	}
+
+	if (config.MaxIdleTime != 0) {
+		sqlDB.SetConnMaxIdleTime(time.Second * time.Duration(config.MaxIdleTime))
+	} else {
+		sqlDB.SetConnMaxIdleTime(time.Minute * 5)
```
Comment (0xmountaintop) [inline common/database/db.go]:
```suggestion
		sqlDB.SetConnMaxIdleTime(defaultConnMaxIdleTime)
```

**Comment by 0xmountaintop** [comment]
CI fails

**Comment by jeswinsimon** [comment]
> CI fails

Resolved.

**Comment by Thegaram** [comment]
Hi @jeswinsimon, what's the rationale for making these values configurable?

**Comment by 0xmountaintop** [comment]
hi @jeswinsimon the CI fails again. can you fix them? so that we can merge it soon

**Comment by georgehao** [comment]
This pr hasn't been updated in a long time, close it. if necessary, pls reopen it

**Pull request closed without merging**